src/main/java/org/squbs/sample/SampleHttpService.java

```diff
@@ -0,0 +1,106 @@
+import akka.http.javadsl.server.PathMatchers;
+import akka.http.javadsl.server.Route;
+import akka.stream.javadsl.Source;
+import akka.util.ByteString;
+import akka.util.Timeout;
+import org.squbs.actorregistry.japi.ActorLookup;
+import org.squbs.unicomplex.AbstractRouteDefinition;
+import scala.concurrent.duration.Duration;
+import scala.concurrent.duration.FiniteDuration;
+
+import java.util.concurrent.TimeUnit;
+
+import static akka.http.javadsl.server.PathMatchers.segment;
+
+/**
+ * The route definition.
+ */
+public class SampleHttpService extends AbstractRouteDefinition {
+
+    final ActorLookup<?> lookup = ActorLookup.create(context());
+    final Timeout timeout = new Timeout(Duration.create(5, TimeUnit.SECONDS));
+
+    private Route anonymous = route(
+        path("hello", () -> pathEnd(() ->
+            onComplete(lookup.ask(new PingRequest("anonymous"), timeout).thenApply(PingResponse.class::cast),
+                response -> {
+                    if (response.isSuccess()) {
+                        return complete(StatusCodes.OK, response.get().message, Jackson.marshaller());
+                    } else {
+                        return complete(StatusCodes.BAD_REQUEST);
+                    }
+                })
+        )));
+
+    private Route withName = route(
+        path(segment("hello").slash(PathMatchers.segment()), who -> pathEnd(() ->
+            onComplete(lookup.ask(new PingRequest(who), timeout).thenApply(PingResponse.class::cast),
+                response -> {
+                    if (response.isSuccess()) {
+                        return complete(StatusCodes.OK, response.get(), Jackson.marshaller());
+                    } else {
+                        return complete(StatusCodes.BAD_REQUEST);
+                    }
+                }
+            )
+        )));
+
+    private Route chunks = route(
+        path(segment("hello").slash(PathMatchers.segment().slash(PathMatchers.integerSegment())), (who, delay) ->
+            // This header is added for Chrome to handle chunking responses.  Please see
+            // http://stackoverflow.com/questions/26164705/chrome-not-handling-chunked-responses-like-firefox-safari
+            respondWithHeader(RawHeader.create("X-Content-Type-Options", "nosniff"), () ->
+
+                onComplete(lookup.ask(
+                    new ChunkRequest(who, FiniteDuration.create(delay, TimeUnit.MILLISECONDS)), timeout)
+                        .thenApply(ChunkSourceMessage.class::cast),
+                    response -> {
+                        if (response.isSuccess()) {
+                            Source<PingResponse, NotUsed> source = response.get().source.concat(
+                                Source.single(new PingResponse(AppConstants.chunkEnd)));
+
+                            return complete(HttpEntities.createChunked(ContentTypes.TEXT_PLAIN_UTF8, source
+                                .map(e -> e.message.equals(AppConstants.chunkEnd) ?
```

Comment:
I think I have just one point. If you do the `map` before the `concat`, the `map` would not have to even check for `AppConstants.chunkEnd`.